advanced/sr_gui_fingertip_visualization/src/sr_gui_fingertip_visualization/dot_unit.py

```diff
@@ -0,0 +1,233 @@
+
+
+class DotUnitGeneric(QWidget):
+    def __init__(self, parent, index=None):
+        super().__init__(parent=parent)
+        self._dot = CircleDot(index=index, parent=self)
+        self._dot.update_color_dot = self._value_to_color
+
+    def resize_dot(self, radius):
+        self._dot._radius = radius
+        self._dot._center = QPoint(self._dot.frameSize().width()/2, self._dot._radius + self._dot._title_height)
+        self._dot.setMinimumSize(self._dot.MIN_SIZE_X, 2*radius)
+
+    def _initialize_data_structure(self, **optional_parameters):
+        raise NotImplementedError
+
+    def _init_widget(self):
+        raise NotImplementedError
+
+    def _value_to_color(self, value):
+        # Overwrite self._dot._color = QColor(r,g,b)
+        raise NotImplementedError
+
+    def get_dot(self):
+        return self._dot
+
+    def update_data(self, value):
+        raise NotImplementedError
+
+
+class DotUnitPST(DotUnitGeneric):
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self._initialize_data_structure()
+        self.resize_dot(25)
+        self._init_widget()
+
+    def _value_to_color(self, value):
+        max_color, min_color = 255, 0
+        value = max(min_color, min(max_color, (value - self._pst_min)*(max_color-min_color) /
+                                   (self._pst_max - self._pst_min) + min_color))
+        self._dot._color = QColor(255-value, 0, value)
+
+    def _initialize_data_structure(self):
+        self._data = dict()
+        self._data_fields = ['pressure', 'temperature']
+        self._pst_max, self._pst_min = 300, 800
+        for data_field in self._data_fields:
+            self._data[data_field] = 0
+
+    def _init_widget(self):
+        main_layout = QVBoxLayout()
+        main_layout.setContentsMargins(0, 0, 0, 0)
+        data_layout = QFormLayout()
+        data_layout.addRow(self.get_dot())
+        self.label = dict()
+
+        for data_field in self._data_fields:
+            self.label[data_field] = QLabel("-")
+            data_layout.addRow(QLabel(str(data_field[0])+":"), self.label[data_field])
+        self.setLayout(data_layout)
+
+    def update_data(self, data):
+        for data_field in self._data_fields:
+            self.label[data_field].setText(str(data[data_field]))
+        self._dot.update_color_dot(data['pressure'])
+        self.get_dot().update()
+
+
+class DotUnitBiotacSPMinus(DotUnitGeneric):
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self._initialize_data_structure()
+        self.resize_dot(20)
+        self._init_widget()
+
+    def _initialize_data_structure(self):
+        self._data_fields = ['pac0', 'pac1', 'pdc', 'tac', 'tdc']
+
+    def _value_to_color(self, value):
+        # to change value (1000 and 200) as its taken from the topic
+        r = min(255, max(0, 255 * (value - 1000) / 200))
+        g = 0
+        b = 255 - r
+        self._dot._color = QColor(r, g, b)
```

Comment:
you are accessing color that should not be accessible. consider adding a setColor method if you need to